lib/RestructureCFG/BeautifyGHAST.cpp

```diff
@@ -1024,6 +1025,11 @@ void beautifyAST(const model::Binary &Model, Function &F, ASTTree &CombedAST) {
   RootNode = inlineDispatcherSwitch(CombedAST);
   Dumper.log("after-dispatcher-switch-inlining");
 
+  // Perform the dead code simplification
```

Comment:
Please add more explanation about why it makes sense to call this here and not any other places.

Reply:
Added some related comments [here](https://github.com/revng/revng-c/blob/6af6c8ffa81d1d570010e6e64e260a78eedd4ea9/lib/RestructureCFG/BeautifyGHAST.cpp#L1028)